BonusCalcApi/V1/UseCase/UpdateWeekUseCase.cs

```diff
@@ -0,0 +1,34 @@
+using System.Threading.Tasks;
+using BonusCalcApi.V1.Boundary.Request;
+using BonusCalcApi.V1.Gateways.Interfaces;
+using BonusCalcApi.V1.Infrastructure;
+using BonusCalcApi.V1.UseCase.Interfaces;
+
+namespace BonusCalcApi.V1.UseCase
+{
+    public class UpdateWeekUseCase : IUpdateWeekUseCase
+    {
+        private readonly IWeekGateway _weekGateway;
+        private readonly IDbSaver _dbSaver;
+
+        public UpdateWeekUseCase(IWeekGateway weekGateway, IDbSaver dbSaver)
+        {
+            _weekGateway = weekGateway;
+            _dbSaver = dbSaver;
+        }
+
+        public async Task<Week> ExecuteAsync(string weekId, WeekUpdate request)
+        {
+            var week = await _weekGateway.GetWeekAsync(weekId);
+
+            if (week is null) return null;
+
+            week.ClosedAt = request.ClosedAt;
```

Comment:
Are we doing a check for if the update has the same data as the original? Is this allowed? Would it be prudent to log a warning?

Reply:
I was thinking about being able to re-open a week by setting it to null - maybe it should only allow changing between the two states? i.e. if closed only allow setting to null and vice-versa.

Reply:
I think that would be prudent as ab extra safeguard yes. The rest looks OK so I'll leave to you if you want to take as is.